cloudify_types/cloudify_types/utils.py

```diff
@@ -67,24 +67,30 @@ def get_desired_operation_input(key, args):
             ctx.node.properties.get(key))
 
 
+def _is_client_external(client, client_config):
+    tenant = client_config.get('tenant')
+    if tenant and tenant != ctx.tenant_name:
+        return True
+
+    manager_ips = set()
+    for mgr in manager.get_rest_client().manager.get_managers():
+        manager_ips |= {mgr.private_ip, mgr.public_ip}
+    internal_hosts = {'127.0.0.1', 'localhost'} | manager_ips
+    host = {client.host} if isinstance(client.host, str) else set(client.host)
+    is_internal = host & internal_hosts
+    if is_internal and (host - internal_hosts):
```

Comment:
I'm also changing this condition, because the previous impl also emitted the warning in case of no match at all. Whoops.

Reply:
Nice catch!